Author: Matthias Klumpp <matthias@tenstral.net>

    Exit with friendly error message when failing to create device nodes

diff --git a/functions b/functions
index 674d649..a0556d8 100644
--- a/functions
+++ b/functions
@@ -80,7 +80,7 @@ wgetprogress () {
 		wget "$@" 2>&1 >/dev/null | $PKGDETAILS "WGET%" $PROGRESS_NOW $PROGRESS_NEXT $PROGRESS_END >&3
 		ret=$?
 	else
-		wget $QSWITCH "$@" 
+		wget $QSWITCH "$@"
 		ret=$?
 	fi
 	return $ret
@@ -1050,7 +1050,8 @@ setup_devices () {
 		setup_devices_hurd ;;
 	    *)
 		if [ -e "$DEVICES_TARGZ" ]; then
-			zcat "$DEVICES_TARGZ" | (cd "$TARGET"; tar -xf -)
+			zcat "$DEVICES_TARGZ" | (cd "$TARGET"; tar -xf -) || \
+					error 1 DEVCREATE "Failed to create devices using %s." "$DEVICES_TARGZ"
 		else
 			if [ -e /dev/.devfsd ] ; then
 				in_target mount -t devfs devfs /dev
